webapp/src/ts/modules/messages/messages-content.component.html

```diff
@@ -57,7 +57,7 @@
       <div id="message-footer"
         class="reply message-form"
         [ngClass]="{sending: textAreaFocused || send.message?.length}"
-        *ngIf="selectedConversation.messages[0] && (selectedConversation.contact?.name || selectedConversation.contact?.doc)">
+        *ngIf="selectedConversation.messages[0] && (selectedConversation.contact?.name || selectedConversation.contact?.doc?.name)">
```

Comment:
Kind of strange that we're looking for a "name" to determine whether we have a "contact" or not. I _think_ it would be more correct to allow replying if we have a phone number to send a message to. 

If the phone number is available, we should allow sending a message to this phone number. 
We, afterall, allow sending messages to phone numbers that are not registered in the system. We should treat these "threads" the same way.

Reply:
I agree but the phone number is assigned to the contact's property `name` when the phone number is not associated with anyone. Here are the 3 different cases:

#### Conversation with a contact associated with someone
```json
{
    "id": "7aeaa4bc-ae5e-4964-8771-697209d2c6dd",
    "messages": [...],
    "contact": {
        "_id": "7aeaa4bc-ae5e-4964-8771-697209d2c6dd",
        "lineage": [...],
        "doc": {
            "_id": "7aeaa4bc-ae5e-4964-8771-697209d2c6dd",
            "_rev": "3-c22e5a626724aa12a02941dd27ec9649",
            "parent": {
                "_id": "87d06670-d877-4d09-aa98-4df98ebf20f4"
            },
            "type": "person",
            "name": "Groot",
            "short_name": "",
            "date_of_birth": "2000-01-31",
            "date_of_birth_method": "",
            "ephemeral_dob": {
                "dob_calendar": "2000-01-31",
                "dob_method": "",
                "ephemeral_months": "7",
                "ephemeral_years": "2021",
                "dob_approx": "2021-07-07",
                "dob_raw": "2000-01-31",
                "dob_iso": "2000-01-31"
            },
            "phone": "+33613370777",
            "phone_alternate": "",
            "sex": "female",
            "role": "chw",
            "external_id": "",
            "notes": "",
            "meta": {
                "created_by": "admin",
                "created_by_person_uuid": "",
                "created_by_place_uuid": "",
                "last_edited_by": "admin",
                "last_edited_by_person_uuid": "",
                "last_edited_by_place_uuid": ""
            },
            "reported_date": 1625561163053,
            "patient_id": "49918"
        }
    }
}
```

#### Conversation with a contact **not** associated with anyone
```json
{
    "id": "+40788123456",
    "messages": [...],
    "contact": {
        "name": "+40788123456"
    }
}
```

#### Conversation with a contact not associated with a deleted person
```json
{
    "id": "52c1de4a-33dd-4d22-9916-24e7c77451f5",
    "messages": [...],
    "contact": {
        "doc": {
            "name": ""
        }
    }
}
```

Reply:
Yes sure. So, in the case of a contact not associated with anyone, we get the phone number from somewhere, right? 
My suggestion was to do the same for deleted contacts, so we allow the UI to still send messages to their phone numbers. Or is that not possible?

Reply:
For deleted contacts, their information is kept in PouchDB in a tombstone. I'm not sure if we're supposed to query this kind of document and how reliable it is. Have we ever dealt with something similar?

Reply:
Don't we have the phone number available in the report (message) doc? As the `from` field or something? 

Tombstones are not available for offline users, so I wouldn't use those for this.

Reply:
This is what the messages array looks like. We do have the `from` phone number in the message's document `messages[0].doc.from` and `messages[0].doc.sms_message.from` but for some reason we are not using it in the `messages[0].from` field. I'm looking into the HydrateMessagesService to see how we can use the data we have.

```json
[
    {
        "doc": {
            "_id": "40b0678a-0f16-47cb-8591-fb3eab3b81d2",
            "_rev": "1-6472f1283ce44dec31e9620ddf87f8c0",
            "type": "data_record",
            "from": "+40788123457",
            "errors": [],
            "tasks": [],
            "fields": {},
            "reported_date": 1625564193569,
            "sms_message": {
                "type": "sms_message",
                "message": "dddd",
                "form": "DDDD",
                "from": "+40788123457"
            },
            "contact": {
                "_id": "52c1de4a-33dd-4d22-9916-24e7c77451f5",
                "parent": {
                    "_id": "2ac7a997-7d6d-44cf-8d70-f243afeb55a0",
                    "parent": {
                        "_id": "87d06670-d877-4d09-aa98-4df98ebf20f4"
                    }
                }
            }
        },
        "id": "40b0678a-0f16-47cb-8591-fb3eab3b81d2",
        "key": "52c1de4a-33dd-4d22-9916-24e7c77451f5",
        "lineage": [],
        "outgoing": false,
        "from": "52c1de4a-33dd-4d22-9916-24e7c77451f5",
        "date": 1625564193569,
        "type": "contact",
        "message": "dddd",
        "read": true
    }
]
```